### PR TITLE
Chore/sof 6463

### PR DESCRIPTION
**Pull request by spectre007**
This PR concerns:
- isolation & refactoring of filter functions
- isolation of other utility functions

### CODE REVIEW

File: src/utils/filter.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+/**
+ * Check if _all_ paths in concatenated path match filtering conditions.
+ * @param {{path: string}} pathObject - Path object with concatenated path (multipath)
+ * @param {string} multiPathSeparator - String sequence used for concatenation of paths
+ * @param {Array<{path: string}|{regex: RegExp}>} filterObjects - Filter conditions
+ * @return {boolean}
+ */
+function isMultiPathSupported(pathObject, multiPathSeparator, filterObjects) {
```
Comment (timurbazhirov) [inline src/utils/filter.js:29]: Maybe use npm https://www.npmjs.com/package/path

File: src/utils/filter.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Check if one path matches regular expression or exact string.
+ * @param {{path: string}} pathObject - Entity or object with path property
+ * @param {Array<{path: string}|{regex: RegExp}>} filterObjects - Filter conditions
+ * @return {boolean}
+ */
+function isPathSupported(pathObject, filterObjects) {
+    return (
+        lodash.find(filterObjects, (filterObj) => {
```
Comment (timurbazhirov) [inline src/utils/filter.js:11]: Nit, but: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some could also work

File: src/utils/filter.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Check if _all_ paths in concatenated path match filtering conditions.
+ * @param {{path: string}} pathObject - Path object with concatenated path (multipath)
+ * @param {string} multiPathSeparator - String sequence used for concatenation of paths
+ * @param {Array<{path: string}|{regex: RegExp}>} filterObjects - Filter conditions
+ * @return {boolean}
+ */
+function isMultiPathSupported(pathObject, multiPathSeparator, filterObjects) {
+    const expandedPaths = pathObject.path.split(multiPathSeparator).map((p) => ({ path: p }));
+    return lodash.every(expandedPaths, (obj) => isPathSupported(obj, filterObjects));
```
Comment (unsigned6) [inline src/utils/filter.js:31]: Could be done without lodash (few lines but the same with JS only:

```js
for (const expandedPath of expandedPaths) {
    if(!isPathSupported(expandedPath, filterObjects) return false;
}
retrun true;    
```

**Review by unsigned6 — COMMENTED** [review]
Looks good

File: src/utils/filter.js (hunk 3 above)
Comment (spectre007) [reply src/utils/filter.js:31]: I used the following now, avoiding lodash and explicit for loops:
```javascript
return expandedPaths.every((expandedPath) => isPathSupported(expandedPath, filterObjects));
```

**Pull request merged**